addons/website/models/ir_module_module.py

```diff
@@ -660,7 +660,7 @@ def get_distinct_snippet_names(structure):
         # Configurator
         # ------------------------------------------------------------
 
-        configurator_snippets = manifest['configurator_snippets']
+        configurator_snippets = manifest.get('configurator_snippets', {})
```

Comment:
How is this possible?
`manifest` is retrieve through `load_manifest()` at the end, which does a `manifest = copy.deepcopy(_DEFAULT_MANIFEST)` and we made sure that `_DEFAULT_MANIFEST` contains a `configurator_snippets` entry

Reply:
Hello @rdeodoo and @bso-odoo 

It is possible . even more then 3-4 database.
how : 
Inside `load_manifest()` function will find mod_path . 
in our of the request , we got `theme_common` 
and for that module we got this path : 
`/home/odoo/.local/share/Odoo/addons/17.0/theme_common`

and that's why we got none value on `manifest_file`
`manifest_file = module_manifest(mod_path)`
and if we have `not manifest_file` then we return empty {}. 

see :  https://github.com/odoo/odoo/blob/912a91817546e92a52df4e23bfd980d9ad4eb698/odoo/modules/module.py#L311C29-L311C29

```
Traceback (most recent call last):
  File "/home/odoo/src/odoo/17.0/odoo/tools/convert.py", line 556, in _tag_root
    f(rec)
  File "/home/odoo/src/odoo/17.0/odoo/tools/convert.py", line 280, in _tag_function
    _eval_xml(self, rec, env)
  File "/home/odoo/src/odoo/17.0/odoo/tools/convert.py", line 205, in _eval_xml
    return odoo.api.call_kw(model, method_name, args, kwargs)
  File "/home/odoo/src/odoo/17.0/odoo/api.py", line 466, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/home/odoo/src/odoo/17.0/odoo/api.py", line 453, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/odoo/src/odoo/17.0/addons/website/models/ir_module_module.py", line 735, in _generate_primary_snippet_templates
    module._generate_primary_snippet_templates()
  File "/home/odoo/src/odoo/17.0/addons/website/models/ir_module_module.py", line 663, in _generate_primary_snippet_templates
    configurator_snippets = manifest['configurator_snippets']
KeyError: 'configurator_snippets'
```

Reply:
I also got this issue, not sure how it started to occur, I changed the code to what @rdeodoo suggest but it then carried on to error at line 668:
`templates = manifest['new_page_templates']`

So not sure if that wants to be updated to a get?

I just haven't managed to figure out what has caused the error. Only thing i can think that i changed was a field from store=False to store=True.

Reply:
This is probably the right fix reading this thread, we will have a look this week :+1: